packages/services/src/kernel/future.ts

```diff
@@ -19,10 +19,11 @@ declare var setImmediate: any;
  * is considered done when the `idle` status is received.
  *
  */
-export class KernelFutureHandler<
-  REQUEST extends KernelMessage.IShellMessage = KernelMessage.IShellMessage,
-  REPLY extends KernelMessage.IShellMessage = KernelMessage.IShellMessage
+export abstract class KernelFutureHandler<
+  REQUEST extends KernelMessage.IShellControlMessage = KernelMessage.IShellControlMessage,
```

Comment:
I think it doesn't make sense to have defaults for these type parameters, since they really should be overridden below.